biojava-core/src/main/java/org/biojava/nbio/core/util/SequenceTools.java

```diff
@@ -104,13 +119,31 @@ public static boolean isNucleotideSequence(String sequence)
 			return true;
 	}
 
+	/**
+	 * Attempts to parse String as a DNA sequence first.<br/>
+	 * If this fails it tries to  parse as a ProteinSequence.
+	 * <br/>
+	 * This method does not attempt to create an RNASequence.
+	 * <p>
+	 * Also, a sequence such as 'ATCGTA' which is both a
+	 * peptide sequence and a DNA sequence, will always be returned 
+	 * as a DNA sequence.
+	 * </p>
+	 * <p>
+	 * An empty string argument returns a ProteinSequence of length 0.
+	 * A null argument throws a {@link NullPointerException}
+	 * @param sequence
+	 * @return Either a DNASequence or a ProteinSequence
+	 * @throws CompoundNotFoundException
+	 */
```

Comment:
Nice! thanks for the docs!

Reply:
Yes.. although whether that behaviour is what is desired though might be another matter... I guess this is a non-standard way to create a new Sequence since the Cookbook/Docs uses more rigorous methods to accomplish that.